### PR TITLE
Robot framework requests library

### PR DESCRIPTION
**Pull request by konopskyyyy**

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+author: Daniel Konopski
+layout: post
+published: false
+post_date: 2017-10-12 14:31:00
+tags: 
+  - Robot Framework 
+  - Request Library
+  - REST API
+categories:
+  - Validation
```
Comment (mek-x) [inline config.yml:11]: no  end of line

File: request_library.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
+# Robot Framework - using Request library for control RPI GPIO's
+
+Requests library is one of the most popular libraries implemented for Robot
+framework. It is very important for testing scattered applications, so this is
+the first library I got to know in the robot framework. The library requires
```
Comment (mek-x) [inline request_library.md:5]: robot framwork raz z małych liter, raz z dużej pierwsze słowo a raz wszystkie

File: request_library.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+In the Rest Api we use methods:
+- Delete
+- Get
+- Options
+- Post
+- Put
+- Patch
+
+For control GPIO's on the RPI we need only `Get` and `Patch`, other methods are
```
Comment (mek-x) [inline request_library.md:22]: To control

File: request_library.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+used in the same way - `Options` are exceptions.
+
+I assume you have already installed Robot Framework. If response is "NO",
+I would like you to invite the paragraph [Installation paragraph](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#installation-instructions)
```
Comment (mek-x) [inline request_library.md:26]: invite you to read the Installation paragraph

File: request_library.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+```
+${URL} http://192.168.0.46:5000
+```
+
+Let's test out first test `Get signals`. Firstly  we should create session 
+for request control in Robot Framework. This convention uses all
+libraries eg. SSH.
+
+For create session we use keyword `Create session` with parameters `alias`, `URl`
```
Comment (mek-x) [inline request_library.md:80]: To create

File: request_library.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+```
+*** Test Cases ***
+
+Get signal
+Create Session gpio_server ${URL} verify=True
+```
+
+We will use `alias` for all keywords which work on URL. All sessions should be
+closed using keyword `Delete All Sessions`.
+Next we need header, which should be send with request. One of the methods for creating
```
Comment (mek-x) [inline request_library.md:92]: should be sent

File: request_library.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+
+Get signal
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pin}= Get Request gpio_server /api/v1/pin/18 ${headers}
+Log  ${pin.json()}
+Delete All Sessions
+```
+
+For getting all signals you can only delete number of pin on hyperlink.
```
Comment (mek-x) [inline request_library.md:139]: Nie rozumiem o co chodzi, coś nie tak z tym zdaniem

File: request_library.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+We will use `alias` for all keywords which work on URL. All sessions should be
+closed using keyword `Delete All Sessions`.
+Next we need header, which should be send with request. One of the methods for creating
+header looks like:
+
+```
+Create Dictionary Content-Type=application/json Accept=application/json
+```
+
+Good trick is assigning headers to variable. For checking value of GPIO is prepared
```
Comment (mek-x) [inline request_library.md:99]: To check the value of the GPIO, special GET method is prepared.

File: request_library.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+Get signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin ${headers}
+Log ${pinout.json()}
+```
+
+That was easier part of our script.
+
+To set values for GPIO's is designed patch method.
```
Comment (mek-x) [inline request_library.md:152]: To set the values for the GPIOs, the patch method needs to be designed (?) - nie do końca rozumiem czy o to chodziło

File: request_library.md [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+Set 0 signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 0
+${pin}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pin.json()}
+Delete All Sessions
+```
+
+The same test we can use for setting high values on GPIO. (?)For indicator I used LED
```
Comment (mek-x) [inline request_library.md:170]: * (?) ?
* We can use the same test to set the high state on GPIO.

File: request_library.md [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+diodes connected serial with resistors.
```
Comment (mek-x) [inline request_library.md:171]: As an indicator, I've used LEDs with resistors in series.

File: request_library.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+Log ${pinout.json()}
+
+Get all signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin ${headers}
+Log ${pinout.json()}
+```
+
+This code hasn't got ideal form but it is not subject of this article.
```
Comment (mek-x) [inline request_library.md:214]: This code is not in perfect form, but it's not the subject of this article. So polishing is left to the reader.

File: request_library.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+After all tests RF generates very readable reports.
```
Comment (mek-x) [inline request_library.md:215]: RF generates very readable reports after all tests are done.

File: request_library.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+
+Reports look like this:
+![](report.png)
+
+You can see that Robot Framework is very easy to use environment for testing REST
+API. All method (e.g. POST and Patch) are the same in using. If you need
```
Comment (mek-x) [inline request_library.md:221]: in use

File: request_library.md [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+help with implementing tests you can ask me or chat with users on slack - request
```
Comment (mek-x) [inline request_library.md:222]: request channel? nie do końca rozumiem

File: request_library.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+channel, when sometimes I help new users or also I am looking for help.
```
Comment (mek-x) [inline request_library.md:223]: I help new users, when I can, and also sometimes I'm looking for help.

**Review by mek-x — CHANGES_REQUESTED** [review]
@konopskyyyy ^

**Comment by konopskyyyy** [comment]
@mek-x dzięki, zrobiłem poprawki i przeniosłem (bo to złe repo) [https://github.com/3mdeb/news-and-ideas/pull/1](https://github.com/3mdeb/news-and-ideas/pull/1)

**Pull request closed without merging**